AX: Fix failing WPT menu-roles test

hs85jeong opened the pull request:
#### 2e6a3871e57140c019de09c3aa2a4fdcfe46a9e0
<pre>
AX: Fix failing WPT menu-roles test
<a href="https://bugs.webkit.org/show_bug.cgi?id=261457">https://bugs.webkit.org/show_bug.cgi?id=261457</a>
rdar://problem/115336495

Reviewed by Tyler Wilcock.

Currently, some of the WPT menu-roles tests[1] have failed. According to
the ARIA spec[2], if a menuitem has a group as first parent and a menu
or menubar as the second parent, it must be a menuitem role.

[1]: <a href="https://wpt.fyi/results/wai-aria/role/menu-roles.html">https://wpt.fyi/results/wai-aria/role/menu-roles.html</a>
[2]: <a href="https://w3c.github.io/aria/#menu">https://w3c.github.io/aria/#menu</a>

* LayoutTests/accessibility/roles-computedRoleString.html:
* LayoutTests/platform/mac/accessibility/roles-computedRoleString-expected.txt:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::remapAriaRoleDueToParent const):

Canonical link: <a href="https://commits.webkit.org/267955@main">https://commits.webkit.org/267955@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/003e237f7993621cf68c604d169ebce68f5707e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18111 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18443 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19010 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19948 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16951 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18308 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21740 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18601 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18916 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18330 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18571 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15763 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20826 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15795 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16516 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23031 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16814 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16687 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20910 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17257 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14619 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16349 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/16357 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4327 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20710 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17105 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hs85jeong commented on LayoutTests/imported/w3c/web-platform-tests/wai-aria/role/menu-roles-expected.txt:
```diff
@@ -0,0 +1,43 @@
+PASS role is menubar
+PASS role is menuitem (in menubar)
+PASS role is group (in menubar)
+PASS role is menuitem (in group, in menubar)
```
https://wpt.fyi/results/wai-aria/role/menu-roles.html
PASS role is menuitem (in group, in menu)
PASS role is menuitem (in group, in menubar)
Two tests pass after this commit.

twilco reviewed (approved):
Thanks for importing the test and for this PR. Will land after EWS completes!

hs85jeong committed:
AX: Fix failing WPT menu-roles test
https://bugs.webkit.org/show_bug.cgi?id=261457
rdar://problem/115336495

Reviewed by Tyler Wilcock.

Currently, some of the WPT menu-roles tests[1] have failed. According to
the ARIA spec[2], if a menuitem has a group as first parent and a menu
or menubar as the second parent, it must be a menuitem role.

[1]: https://wpt.fyi/results/wai-aria/role/menu-roles.html
[2]: https://w3c.github.io/aria/#menu

* LayoutTests/accessibility/roles-computedRoleString.html:
* LayoutTests/platform/mac/accessibility/roles-computedRoleString-expected.txt:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::remapAriaRoleDueToParent const):

Canonical link: https://commits.webkit.org/267955@main
